src/crawlee/events/_event_manager.py

```diff
@@ -125,22 +135,41 @@ async def __aexit__(
         await self._emit_persist_state_event_rec_task.stop()
         self._active = False
 
-    def on(self, *, event: Event, listener: Listener) -> None:
+    @overload
+    def on(self, *, event: Literal[Event.PERSIST_STATE], listener: EventListener[EventPersistStateData]) -> None: ...
+    @overload
+    def on(self, *, event: Literal[Event.SYSTEM_INFO], listener: EventListener[EventSystemInfoData]) -> None: ...
+    @overload
+    def on(self, *, event: Literal[Event.MIGRATING], listener: EventListener[EventMigratingData]) -> None: ...
+    @overload
+    def on(self, *, event: Literal[Event.ABORTING], listener: EventListener[EventAbortingData]) -> None: ...
+    @overload
+    def on(self, *, event: Literal[Event.EXIT], listener: EventListener[EventExitData]) -> None: ...
+    @overload
+    def on(self, *, event: Event, listener: EventListener[None]) -> None: ...
```

Comment:
Is this for user defined Events or why is Event not one of the Literal[Event....]

Reply:
That should mean "anything not matched by the more specific overloads"

Reply:
Ok, I see that **Session pool events** and **Browser pool events** have no event data connected to them. But this overload allows even events with related event data to be called with EventListener[**None**]. For example call like this:

on(event: Literal[Event.EXIT], listener: EventListener[**None**])

Is that intentional?

Reply:
Not exactly intentional, but also not a problem I believe. Who would make a listener that is annotated to accept a literal None?

Reply:
Well, `EventListener[None]` allows these two (plus their async counterparts):

```py
def listener(weird: None) -> None:  # No reason to do this
    print("Ding")

def parameterless_listener() -> None:  # The encouraged way
    print("Ding")
```